front-end/src/app/shared/interfaces/transaction-type.interface.ts

```diff
@@ -19,5 +19,6 @@ export interface TransactionType {
   subTransactionTypes?: ScheduleATransactionTypes[] | ScheduleBTransactionTypes[]; // TransactionTypes to choose from when creating a sub transaction
   navigationControls?: TransactionNavigationControls;
   generatePurposeDescription?(): string; // Dynamically generates the text in the CPD or EPD field
+  purposeDescriptionUserInputRequired?: boolean;
```

Comment:
@dheitzer Is this flag somewhat redundant with the isDescriptionSystemGenerated() method? Is there a nuance I'm missing as why that method can't me used instead to determine if a user can edit the Contribution Purpose Description field?

Reply:
```
     system generated
           /      \
         yes       no
                  /   \
  optional          required
and editable.     and editable

It does need to be separate, but rather than a property, we could just have a common function that determines it's value from the spec rather than the class.  that means one less configuration to implement in the typescript

Reply:
this could be implemented in transactionbasecomponent.ts
```
  isDescriptionRequired(transactionType?: TransactionType): boolean {
    return transactionType?.schema.required.includes('contribution_purpose_descrip') || false;
  }

Reply:
@mjtravers the key usage in the code is ```
{{ descriptionIsSystemGenerated ? '(SYSTEM-GENERATED)' : 
          descriptionRequiresUserInput ? '' : '(OPTIONAL)' }}</label
```
@toddlees some suggested avoiding [functions in a templates](https://medium.com/showpad-engineering/why-you-should-never-use-function-calls-in-angular-template-expressions-e1a50f9c0496) so we might want to make this an initialized property?

Reply:
@toddlees some suggested avoiding [functions in a templates](https://medium.com/showpad-engineering/why-you-should-never-use-function-calls-in-angular-template-expressions-e1a50f9c0496) so we might want to make this an initialized property?